### PR TITLE
[21136] Add XML configuration for FlowControllerDescriptor to 2.x

### PR DESCRIPTION
**Pull request by Mario-DL**
## Description

This PR is a backport from #782 to `2.x`. It adds the XML documentation for flow controller descriptors.

@Mergifyio backport 2.13.x 2.10.x 2.6.x

## Contributor Checklist

- [X] Commit messages follow the project guidelines. 
- [X] Code snippets related to the added documentation have been provided.
- [X] Documentation tests pass locally.
- [X] Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] CI passes without warnings or errors.

### CODE REVIEW

**Comment by Mario-DL** [comment]
@richiprosima please test_3 docs

File: docs/fastdds/use_cases/large_data/large_data.rst [hunk 1]
```diff
@@ -258,8 +258,12 @@ Example configuration
 
    .. tab:: XML
 
-      There is currently no way of configuring flow controllers with XML.
-      This will be added in future releases of the product.
+      .. literalinclude:: /../code/XMLTester.xml
+         :language: xml
+         :start-after: <!-->CONF-QOS-FLOWCONTROLLER
+         :end-before: <!--><-->
+         :lines: 2-3, 5-25
```
Comment (JesusPoderoso) [inline docs/fastdds/use_cases/large_data/large_data.rst:265]: We missed the data writer configuration in the original PR too ([see this PR generated docs]() and [the original PR docs]()).
I am fixing the master snippet here:
- #794

```suggestion
         :lines: 2-3, 5-
```

**Comment by JesusPoderoso** [comment]
@richiprosima please test this

**Review by JesusPoderoso — APPROVED** [review]
LGTM with green CI

**Comment by EduPonz** [comment]
@Mergifyio backport 2.13.x 2.10.x 2.6.x

**Pull request merged**